### PR TITLE
Fix 3676

### LINKED ISSUE #3676: only render pan and zoom tools for image frames
## Expected behavior
The pan and zoom controls should only render for subject frames of the type image. The buttons are not functional on other subject frame types (e.g., video, text).

## Current behavior
_Please include any error messages from the browser console and/or screenshots_
The pan and zoom controls are rendered for all frames of a subject that is attached to a workflow with pan and zoom enabled. 

## Steps to replicate
Create workflow with pan and zoom enabled. Attach a subject set with both images and text subject frames. 

## Additional information

- **Operating system:**
- **Browser:**

### PR DESCRIPTION
**Pull request by simensta**
Fixes #3676. 

Describe your changes.
- The PanZoom buttons and eventListeners only render for subject frames of the type 'image'.
- For the TextViewer, `#onLoad()` is now called using a `ref`. In `componentDidMount` and `componentWillMount`, `ReactDOM.findDOMNode(this)` was returning null. The error from [`ReactDOM.findDOMNode(this).addEventListener('load', this.props.onLoad)` on line 31](https://github.com/zooniverse/Panoptes-Front-End/compare/fix-3676?expand=1#diff-f8270d0c4ee56789ba85382262b11d88L31) prevented classification submission for text subject frames.

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] Did you deploy a staging branch? https://fix-3676.pfe-preview.zooniverse.org/
- [ ] Did you get any type checking errors from [Babel Typecheck](https://github.com/codemix/babel-plugin-typecheck)

## Optional

- [ ] If it's a new component, is it in ES6? Is it clear of warnings from ESLint?
- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?

### CODE REVIEW

File: app/components/frame-viewer.cjsx [hunk 1]
```diff
@@ -50,7 +50,7 @@ module.exports = React.createClass
         <TextViewer  src={src} type={type} format={format} frame={@props.frame} onLoad={@handleLoad} /> 
 
     if FrameWrapper
-      <PanZoom ref="panZoom" enabled={zoomEnabled} frameDimensions={@state.frameDimensions}>
+      <PanZoom ref="panZoom" enabled={zoomEnabled} frameType={type} frameDimensions={@state.frameDimensions}>
```
Comment (eatyourgreens) [inline app/components/frame-viewer.cjsx]: Should this render if `type` isn't image?

File: app/components/pan-zoom.jsx [hunk 2]
```diff
@@ -41,12 +42,26 @@ const PanZoom = React.createClass({
   componentDidMount() {
     // these events enable a user to navigate an image using arrows, +, and - keys,
     // while the user is in pan and zoom mode.
-    if (this.props.enabled) {
+    if (this.props.enabled && this.props.frameType === 'image') {
```
Comment (eatyourgreens) [inline app/components/pan-zoom.jsx]: Maybe drop the `frameType` prop and just set `enabled` to false in the parent?

File: app/components/pan-zoom.jsx [hunk 3]
```diff
@@ -77,7 +92,7 @@ const PanZoom = React.createClass({
     return (
       <div>
         {children}
-        {this.props.enabled ?
+        {(this.props.enabled && this.props.frameType === 'image') ?
```
Comment (eatyourgreens) [inline app/components/pan-zoom.jsx]: Again, why not just set `enabled` to false in the parent?

**Comment by eatyourgreens** [comment]
I would enable/disable the component in the frame viewer, based on subject type, rather than adding a new prop to check.

Alternatively, we could rewrite the frame viewer so that pan-zoom isn't rendered except for images.

A subject's `onLoad` handler shouldn't be called until after the subject has loaded, but this change invokes the callback when `componentWillMount` runs. I think it's also running the callback even if there's an error loading the subject. The reason for this is to disable the classification interface so that Done can't be pressed unless the subject was fully loaded.

**Comment by simensta** [comment]
Good point. Conditionally rendering either the PanZoom component or the PanZoom controls from the FrameViewer is preferable to passing another prop to PanZoom. 

I didn't fully understand the original use case for `onLoad`, now I see why this solution doesn't work. 

Since the TextViewer's undefined DOM node is being solved in PR #3689 and the approach to rendering PanZoom will be different, I am going to close this PR and open a new one for #3676.

**Pull request closed without merging**